src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -184,14 +191,19 @@ async def load_lockfile(
     is_pex_native = is_probably_pex_json_lockfile(lock_bytes)
     if is_pex_native:
         header_delimiter = "//"
+        stripped_lock_bytes = _strip_comments_from_pex_json_lockfile(lock_bytes)
         lockfile_digest = await Get(
             Digest,
-            CreateDigest(
-                [FileContent(lockfile_path, _strip_comments_from_pex_json_lockfile(lock_bytes))]
-            ),
+            CreateDigest([FileContent(lockfile_path, stripped_lock_bytes)]),
         )
         requirement_estimate = _pex_lockfile_requirement_count(lock_bytes)
         constraints_strings = None
+        if request.parse_lockfile:
+            try:
+                parsed_lockfile = json.loads(stripped_lock_bytes)
```

Comment:
Are all lockfiles going to be JSON objects? That seems like a big assumption.

Reply:
This is only for parsing the pex lockfile, so doesn't feel like such an big assumption. Still, the format/schema may change over time, but we can adapt when that time comes.

For other languages/eco systems (java/scala/go/etc) there would be similar parsing needed to be implemented.